open-apparel-registry/open-apparel-registry
# Implement Source model - Step 1

## Overview

The first step of a three step process to add a Source model between facility lists and list items, allowing for the submission of individual facilities.  

Each step in the process is intended to be released separately and maintain backward compatibility.

Connects #817 

## Testing Instructions

* Checkout `develop` and `./scripts/resetdb`
* Checkout this branch and `./scripts/migrate`
* Ensure that `./scripts/manage liststosources` runs without error
* Use `./scripts/manage shell_plus` to verify that a `Source` is created for each `FacilityList` and that a `Source` is assigned on each `FacilityListItem`
```
>>> list(FacilityList.objects.all().order_by('id').values_list('id', flat=True))
[2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15]
>>> list(Source.objects.all().order_by('facility_list').values_list('facility_list', flat=True))
[2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15]
>>> FacilityListItem.objects.filter(source=None).count()
0
```

## Checklist
